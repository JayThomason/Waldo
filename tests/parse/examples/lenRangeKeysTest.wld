/** VERSION 2
 * Checks that can compose len, range, and keys
*/


PingPong

Endpoint Ping;
Endpoint Pong;


Sequences
{
}

Shared
{
}


/**** Ping endpoint *********/
Ping
{

    
    Public Function initiateSend()
    {
        List (element: Number) mList = keys(range(0,10,-1));
        Number someNum = len (mList);
        someNum = len(range(0,10,-1));

        someNum = len('hi');

        mList = keys(range(len([1,2,3]),2));

        List(element:Text) som = keys({});

        //shoud cause error.
        List(element:Text) som2 = keys({1:2});
    }
}


Pong
{
}
