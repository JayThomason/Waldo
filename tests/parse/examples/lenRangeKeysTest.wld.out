*************************
Type mismatch for variable named "som2".  Declared with type [{"ElementType": 
"Text", "Type": "List"}], but assigned to type [{"ElementType": "Number", 
"Type": "List"}]. 
-------
Line numbers:
39
-------
Program text:



36            List(element:Text) som = keys({});
37    
38            //shoud cause error.
39*           List(element:Text) som2 = keys({1:2});
40        }
41    }
42    
43    
*************************



